macros/src/main/scala/org/scalarelational/compiletime/QueryMacros.scala

```diff
@@ -0,0 +1,124 @@
+package org.scalarelational.compiletime
+
+import org.scalarelational.SelectExpression
+import org.scalarelational.instruction.Query
+import org.scalarelational.table.Table
+
+import scala.annotation.compileTimeOnly
+import scala.reflect.macros.blackbox
+
+/**
+ * @author Matt Hicks <matt@outr.com>
+ */
+@compileTimeOnly("Enable macro paradise to expand macro annotations")
+object QueryMacros {
+  def to1[R](c: blackbox.Context)
+            (table: c.Expr[Table])
+            (implicit r: c.WeakTypeTag[R]): c.Expr[Query[Vector[SelectExpression[_]], R]] = {
+    import c.universe._
+
+    val query = c.prefix.tree match {
+      case Apply(_, List(qry)) => qry
+    }
+
+    val instance = typeWrapper[R](c)(weakTypeOf[R], table)
+
+    val conv = q"""
+       val converter = new org.scalarelational.instruction.ResultConverter[$r] {
```

Comment:
Not necessarily, but I think it makes the code a bit cleaner and will help debugging.